graph/src/blockchain/polling_block_stream.rs

```diff
@@ -229,7 +232,13 @@ where
 
         // Get pointers from database for comparison
         let head_ptr_opt = ctx.chain_store.chain_head_ptr()?;
-        let subgraph_ptr = ctx.subgraph_store.block_ptr()?;
+        let mut subgraph_ptr = self.current_block.clone();
+        // If current_block is None then use the subgraph block_ptr.
+        if self.current_block.is_none() {
+            if let Some(block_ptr) = self.subgraph_store.block_ptr()? {
```

Comment:
This should not be fetched here but rather in `new_polling_block_stream`

Reply:
new just doesn't feel like a good place to return a Result. Could alternatively request a BlockPtr as an argument and force this to be retrieve between calling new?

Reply:
`new_polling_block_stream` already returns a result.

Reply:
ah yes you're right, I thought you were referring to "new" :)

Reply:
I guess in this case I'll get rid of the Option completely since it's not really needed anymore 🤔

Reply:
Perhaps ideally `new_block_stream` would receive this input, but to cover the firehose case it would need to contain both the block ptr and the firehose cursor.

Reply:
The option is needed because `None` represents no block ever having been processed (if the subgraph has just been deployed for example).